samcli/local/apigw/authorizers/lambda_authorizer.py

```diff
@@ -284,3 +296,166 @@ def _parse_identity_sources(self, identity_sources: List[str]) -> None:
                     self._identity_sources.append(identity_source_validator)
 
                     break
+
+    def is_valid_response(self, response: str, method_arn: str) -> bool:
+        """
+        Validates whether a Lambda authorizer request is authenticated or not.
+
+        Parameters
+        ----------
+        response: str
+            JSON string containing the output from a Lambda authorizer
+
+        Returns
+        -------
+        bool
+            True if the request is properly authenticated
+        """
+        try:
+            json_response = json.loads(response)
+        except ValueError:
+            raise InvalidLambdaAuthorizerResponse(
+                f"Authorizer {self.authorizer_name} return an invalid response payload"
+            )
+
+        if self.payload_version == LambdaAuthorizer.PAYLOAD_V2 and self.use_simple_response:
+            return self._validate_simple_response(json_response)
+
+        return self._validate_iam_response(json_response, method_arn)
+
+    def _validate_iam_response(self, response: dict, method_arn: str) -> bool:
+        """
+        Helper method to validate if a Lambda authorizer response is valid and authorized
+
+        Parameters
+        ----------
+        response: dict
+            JSON object from Lambda authorizer output
+        method_arn: str
+            The method ARN of this request
+
+        Returns
+        -------
+        bool
+            True if the request is authorized
+        """
+        # validate there exists a principalId (unique ID representing the authenticated request)
+        if not response.get(_RESPONSE_PRINCIPAL_ID):
```

Comment:
All the 'ifs' here make it seem that its ripe for a refactor. Thought: how could this be more maintainable?

Reply:
Yeah, I've thought about that too. We could revisit the same design that was used to validate identity sources (`for validator in list_of_validators: validator.check()`) but we would have the same kind of bloat, however it would be cleaner to read and modify in the future.

Reply:
This may be out of scope for the purposes of this PR, but we could also look towards writing a JSON schema validator, or using an existing library, to replace any of the JSON schema parsing that happens in our local commands (or anywhere for that matter)